runtime/src/main/java/io/quarkus/hazelcast/client/HazelcastClientProducer.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.hazelcast.client;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.HazelcastInstance;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+@ApplicationScoped
+public class HazelcastClientProducer {
+    private HazelcastClientConfig hazelcastClientConfig;
+
+    @Produces
```

Comment:
In the quarkus guide, we used `@Singleton`. Otherwise, it recreating hazelcast-client object in every request. One other thing that I could not figure out is how to eagerly initialize hazelcast-client object. In the guide, it always initiate in the first request which makes it slow.

Reply:
I know, there's some extra quarkus infrastructure needed to push all of that to build-time. To be figured out

Reply:
Also, I believe that the creation of bean for each request was because of the controller being request-scoped.